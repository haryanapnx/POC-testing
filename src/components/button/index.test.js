import * as React from "react";
import { fireEvent, render, screen } from "@testing-library/react";

import MyButton from "./index";

describe("Button", () => {
  test("renders correctly", () => {
    render(<MyButton>Button</MyButton>);
    const button = screen.getByRole("button");

    expect(button).toBeInTheDocument();
  });

  test("Snapshot", () => {
    const { container } = render(<MyButton>Button</MyButton>);

    expect(container).toMatchSnapshot();
  });

  test("onClick props", () => {
    const onClick = jest.fn();
    render(<MyButton onClick={onClick}>Button</MyButton>);

    fireEvent.click(screen.getByRole("button"));

    expect(onClick).toHaveBeenCalledTimes(1);
  });

  test("disabled state", () => {
    const onClick = jest.fn();
    render(
      <MyButton id="my-button" disabled={true} onClick={onClick}>
        Button
      </MyButton>
    );

    expect(screen.getByRole("button")).toBeDisabled();

    fireEvent.click(screen.getByRole("button"));

    expect(onClick).toHaveBeenCalledTimes(0);
  });
});
