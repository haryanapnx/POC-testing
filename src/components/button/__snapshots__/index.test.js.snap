// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot 1`] = `
<div>
  <button
    class="my-button"
  >
    Button
  </button>
</div>
`;
